pkg/planner/core/find_best_task.go

```diff
@@ -1456,11 +1461,16 @@ func findBestTask4LogicalDataSource(lp base.LogicalPlan, prop *property.Physical
 			// even enforce hint can not work with this.
 			return base.InvalidTask, 0, nil
 		}
+		// cache the physical for indexJoinProp
+		defer func() {
+			ds.StoreTask(prop, t)
+			err = validateTableSamplePlan(ds, t, err)
```

Comment:
Do we need this? I think there will never be a `TableSample` in this case.

Reply:
good catch, make sense